homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,174 @@
+"""Flo device object."""
+from datetime import datetime, timedelta
+import logging
+from typing import Any, Callable, Dict, List
+
+from aioflo.api import API
+
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import SIGNAL_DEVICE_UPDATED, SIGNAL_WATER_CONSUMPTION_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FloDevice:
```

Comment:
Why not use the DataUpdateCoordinator ?

Reply:
will do

Reply:
done, kept the same device model but made it extend DataUpdateCoordinator. LMK if this is ok.